test/integration_tests/conftest.py

```diff
@@ -93,6 +110,7 @@ def temp_hub_dir(tmpdir, pytestconfig):
         torch.hub.set_dir(tmpdir)
         yield
         torch.hub.set_dir(org_dir)
+    shutil.rmtree(tmpdir)
```

Comment:
Can we move this to a separate PR as this is a bug fix?

When doing so, can we
1. Replace `tmpdir` with `tmp_path` as it is recommended.
2. Indent this line so that we do not change the behavior when `--use-tmp-hub-dir`.
3. Instead of setting `tmpdir`, maybe create a sub directory, say called `hub`.
4. Update other fixtures that handle assets from download.pytorch.org/torchaudio, so that they do not directly deal with `tmp_path`.